swiftide-integrations/src/anthropic/simple_prompt.rs

```
@@ -1,41 +1,72 @@
+use async_anthropic::{
+    errors::AnthropicError, errors::CreateMessagesError, types::CreateMessagesRequestBuilder,
+};
 use async_trait::async_trait;
-use swiftide_core::SimplePrompt;
+use swiftide_core::{chat_completion::errors::LanguageModelError, indexing::SimplePrompt};
 
 use super::Anthropic;
 
 #[async_trait]
 impl SimplePrompt for Anthropic {
     #[tracing::instrument(skip_all, err)]
-    async fn prompt(&self, prompt: swiftide_core::prompt::Prompt) -> anyhow::Result<String> {
+    async fn prompt(
+        &self,
+        prompt: swiftide_core::prompt::Prompt,
+    ) -> Result<String, LanguageModelError> {
         let model = &self.default_options.prompt_model;
 
         let request = CreateMessagesRequestBuilder::default()
             .model(model)
             .messages(vec![prompt.render().await?.into()])
-            .build()?;
+            .build()
+            .map_err(|e| LanguageModelError::PermanentError(e.into()))?;
 
         tracing::debug!(
             model = &model,
-            messages = serde_json::to_string_pretty(&request)?,
+            messages = serde_json::to_string_pretty(&request)
+                .map_err(|e| LanguageModelError::PermanentError(e.into()))?,
             "[SimplePrompt] Request to anthropic"
         );
 
-        let response = self.client.messages().create(request).await?;
+        let response = self.client.messages().create(request).await.map_err(|e| {
+            let CreateMessagesError::AnthropicError(e) = e;
+            match e {
+                AnthropicError::NetworkError(_) => LanguageModelError::TransientError(e.into()),
+                AnthropicError::UnexpectedError => {
+                    LanguageModelError::PermanentError("Anthropic unexpected error".into())
+                    // Seriously?
+                }
+                AnthropicError::Unauthorized => {
+                    LanguageModelError::PermanentError("Anthropic unauthorized".into())
```

Also on 37, this hides the underlying error. Wouldn't it be better to always wrap the underlying error?